config/envoyconfig/listeners.go

```diff
@@ -133,6 +133,8 @@ func (b *Builder) buildMainListener(
 		li.ListenerFilters = append(li.ListenerFilters, ProxyProtocolFilter())
 	}
 
+	li.AccessLog = buildAccessLogs(cfg.Options)
```

Comment:
I think we need a separate handler for listener access logs, as they appear to be scoped to the connection rather than the request, so most of the existing access log fields will be empty/zero. For example:

> {"level":"info","service":"envoy","upstream-cluster":"","method":"METHOD_UNSPECIFIED","authority":"","path":"","user-agent":"","referer":"","forwarded-for":"","request-id":"","duration":0,"size":1947,"response-code":0,"response-code-details":"","time":"2024-08-01T16:12:12-07:00","message":"http-request"}